### DISCUSSION TITLE
Network size reduction using partial meth topology

### DISCUSSION THREAD

**Discussion by kanekosh** [google/or-tools#3527] [Routing (and legacy CP) questions] [answered]
Would it be possible to use partial mesh topology to reduce the network size for routing?
By the partial mesh topology, I mean that the arcs (edges) are defined between *some* pairs of nodes, but not all pairs. An example of the topology can be found [here](https://networklessons.com/cisco/ccna-routing-switching-icnd1-100-105/network-topologies). In contrast, the [routing example](https://developers.google.com/optimization/routing/vrp) of the documentation uses full mesh topology as it defines the distance matrix and distance callback for every pair of nodes.

The partial mesh topology may results in an inferior solution compared to the full mesh, but it can be beneficial for large-scale problems because it reduces the network size hence the memory requirement and computational time for the solver. 

Thank you in advance!

**Comment by lperron** [comment, marked as answer]
VRP is not TSP. VRP expect dense graphs. TSP profits from sparse graph.

The routing library is a VRP solver that can do TSP. So reduction the number of arcs in the graph will simply make the problem much harder.

**Reply by kanekosh** [reply]
Thank you for your reply.
